packages/gatsby-plugin-preload-fonts/src/prepare/index.js

```diff
@@ -0,0 +1,114 @@
+#! /usr/bin/env node
+
+const path = require(`path`)
+const crypto = require(`crypto`)
+const { URL } = require(`url`)
+
+const puppeteer = require(`puppeteer`)
+const ProgressBar = require(`progress`)
+const { blue, green, bold, dim } = require(`chalk`)
+
+const createLogger = require(`./logger`)
+const fetchRoutes = require(`./fetch-routes`)
+const { load, save, getPath: getCachePath } = require(`./cache`)
+const { ellipses } = require(`./utils`)
+
+const {
+  PORT = 8000,
+  HOST = `localhost`,
+  LOG_LEVEL = `error`,
+  GRAPHQL_PATH = `/___graphql`,
+} = process.env
+
+const logger = createLogger(LOG_LEVEL)
+
+const endpoint = path => `http://${HOST}:${PORT}${path}`
+
+async function main() {
+  logger.newline()
+  logger.info(`loading cache`)
+  const cache = await load()
+
+  logger.info(`requesting route listing`)
+  const routes = await fetchRoutes({
+    logger,
+    cache,
+    endpoint: endpoint(GRAPHQL_PATH),
+  })
+
+  const sections = [
+    dim(` crawling routes`),
+    `:bar`,
+    bold(blue(`:percent`)),
+    dim(`eta :etas`),
+    green(`:last`),
+  ]
+  const bar = new ProgressBar(sections.join(` `), {
+    width: 35,
+    total: routes.length,
+    complete: `█`,
+    incomplete: ` `,
+  })
+  logger.setAdapter((...args) => bar.interrupt(args.join(` `)))
+
+  const browser = await puppeteer.launch()
+  const page = await browser.newPage()
+  let last = ``
+  page.on(`requestfinished`, e => {
+    if (!e._url.match(/(socket\.io|commons\.js)/)) {
+      logger.info(`load`, e._method, e._url)
+    }
+
+    if (
```

Comment:
Checkign resourcetype is probably more robust than checking urls as we rely on chrome's internal api to see if something is a font.
```suggestion
if (request.resourceType() === 'font') {
```
https://pptr.dev/#?product=Puppeteer&version=v1.10.0&show=api-requestresourcetype

Reply:
Nice. I didn't realize the other request properties had nice getter methods (`req.url()`, `req.method()`) so I think I'll switch to using those as well.